src/maplibre_helpers.ts

```diff
@@ -78,15 +86,19 @@ export function overwriteSource(map, id, source) {
   map.addSource(id, source);
 }
 
-export function overwriteLayer(map, layer, beforeId: string | null = null) {
+export function overwriteLayer(
+  map: Map,
+  layer,
```

Comment:
You'll note I'm not saying `LayerSpecification` here yet, because the errors absolutely explode :) One of the hard things remaining